internal/namespace/manager.go

```diff
@@ -124,3 +139,129 @@ func (nm *namespaceManager) initNamespaces(ctx context.Context, di database.Plug
 	}
 	return nil
 }
+
+func (nm *namespaceManager) validateNamespaceConfig(ctx context.Context, name string, index int, conf config.Section) error {
+	if err := core.ValidateFFNameField(ctx, name, fmt.Sprintf("namespaces.predefined[%d].name", index)); err != nil {
+		return err
+	}
+
+	if strings.ToLower(name) == "ff_system" || strings.ToLower(conf.GetString(coreconfig.NamespaceRemoteName)) == "ff_system" {
+		return i18n.NewError(ctx, coremsgs.MsgFFSystemReservedName)
+	}
+
+	mode := conf.GetString(coreconfig.NamespaceMode)
+	plugins := conf.GetStringSlice(coreconfig.NamespacePlugins)
+
+	// If the no plugins are found when querying the config, assume older config file
+	if len(plugins) == 0 {
+		// Still check to ensure at least one bc, dx, ss, and db plugin exist
```

Comment:
In this branch maybe we want to actually iterate all the defined plugins and add them to the `plugins` list, then let it go through the normal validation switch-case below?